developer-docs-site/docs/standards/fungible-asset.md

```diff
@@ -0,0 +1,451 @@
+The final goal of FA is to replace Aptos coin.
+
+## Structures
+
+### Metadata Object
+
+FA metadata defines the type of FA. A metadata is defined in Move as:
+
+```rust
+#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+struct Metadata has key {
+    supply: Option<Supply>,
+    /// Name of the fungible metadata, i.e., "USDT".
+    name: String,
+    /// Symbol of the fungible metadata, usually a shorter version of the name.
+    /// For example, Singapore Dollar is SGD.
+    symbol: String,
+    /// Number of decimals used for display purposes.
+    /// For example, if `decimals` equals `2`, a balance of `505` coins should
+    /// be displayed to a user as `5.05` (`505 / 10 ** 2`).
+    decimals: u8,
+}
+```
+
+:::tip
+This refers to a type system that uses metadata to differentiate between different types, much like CoinType in a coin standard. However, in this system, even if two metadata objects are exactly the same, they represent two distinct types rather than just one.
+:::
+
+A Coin uses the `CoinType` to support re-usability of the Coin framework for distinct Coins. For example, `Coin<A>`
+and `Coin<B>` are two distinct coins.
+
+### Fungible Asset and Fungible Store
+
+Since metadata is specified, FA could be defined as follows:
+
+```rust
+struct FungibleAsset {
+    metadata: Object<Metadata>,
+    amount: u64,
+}
+```
+
+Simple, right? An object representing the type and the amount of units held. It is noted that it does not have any
+abilities, so it can be passed from one function to another but has to be deposited back into a fungible store
+at the end of the transaction. In other words, it must be consumed and cannot be
+directly stored anywhere. But how to store it? Here comes `FungibleStore` for storing them in objects:
+
+```rust
+#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct FungibleStore has key {
+    /// The address of the base metadata object.
+    metadata: Object<Metadata>,
+    /// The balance of the fungible metadata.
+    balance: u64,
+    /// Fungible Assets transferring is a common operation, this allows for freezing/unfreezing accounts.
+    allow_ungated_balance_transfer: bool,
+}
+```
+
+The only extra field added here is `allow_ungated_balance_transfer`. if it is `true`, this object is frozen, i.e.
+deposit and withdraw are both disabled without using `TransferRef` in the next section.
+
+### References
+
+Reference(ref) is the means to implement permission control across different standard in Aptos. In different contexts,
+it may be called capabilities sometimes. In FA standard, there are three refs for Mint, Transfer and Burn operations on
+FA of the same metadata specified in the ref struct.
```

Comment:
```suggestion
Reference(ref) is the means to implement permission control across different standard in Aptos. In different contexts, it may be called capabilities sometimes. In FA standard, there are three distinct refs one each for minting, transferring, and burning FA respectively called `MintRef`, `TransferRef`, and `BurnRef`. Each ref contains a reference to the FA metadata:
```